userinterface/Services/ViewModelFactory.cs

```diff
@@ -0,0 +1,46 @@
+using System;
+using userinterface.Services;
+using userinterface.ViewModels.Profile;
+using userinterface.ViewModels.Device;
+using userinterface.ViewModels.Mapping;
+using BE = userspace_backend.Model;
+
+namespace userinterface.Services
+{
+    public class ViewModelFactory : IViewModelFactory
+    {
+        private readonly INotificationService notificationService;
+        private readonly IThemeService themeService;
+
+        public ViewModelFactory(INotificationService notificationService, IThemeService themeService)
+        {
+            this.notificationService = notificationService ?? throw new ArgumentNullException(nameof(notificationService));
+            this.themeService = themeService ?? throw new ArgumentNullException(nameof(themeService));
+        }
+
+        public ProfileViewModel CreateProfileViewModel(BE.ProfileModel profileModel)
```

Comment:
For "true" dependency injection, this factory should probably be calling the IServiceProvider instead of calling "new". The viewmodels would be bound so that resolving creation of the root class creates the rest and then the root class could have function to take BE.ProfileModel.
This is not a first-priority requirement but I thought I'd point it out. Can resolve comment once read